action/protocol/execution/evm/evm.go

```diff
@@ -203,7 +203,7 @@ func ExecuteContract(
 	receipt := &action.Receipt{
 		GasConsumed:     ps.gas - remainingGas,
 		BlockHeight:     blkCtx.BlockHeight,
-		ActionHash:      execution.Hash(),
+		ActionHash:      actionCtx.ActionHash,
```

Comment:
this is a wrong programming: should use action hash passed from context

and this is why SetEnvelopeContext() currently do:
```
act.hash = Selp.Hash()
```
which is not needed once this is corrected